### PR TITLE
ES6 declaration

### PR DESCRIPTION
**Pull request by newash**
I'm not sure if I'm doing this pull request in the right way, however I'd be happy if you accept it. It's a small change as you see, that eliminates the old "require" syntax.

### CODE REVIEW

**Comment by chbrown** [comment]
@dt-bot: fine by me, though I'm not sure what the DT policy is on TypeScript version backwards compatibility.

@newash: I'm not someone who can merge your changes into DT proper, but it'd be nice to whoever is to squash your commits, since they're each just single line changes.

**Comment by newash** [comment]
@chbrown I second that, it's not the nicest this way (I was doing the whole thing through the web interface, and I don't know any way to modify multiple files in one commit here).

File: object-assign/object-assign.d.ts [hunk 1]
```diff
@@ -4,6 +4,5 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare module "object-assign" {
-  function objectAssign(target: any, ...sources: any[]): any;
-  export = objectAssign;
+  export default function(target: any, ...sources: any[]): any;
```
Comment (mhegazy) [inline object-assign/object-assign.d.ts]: this is not correct. this package does not export a default. it is a CJS module, as per the documentation, see https://github.com/sindresorhus/object-assign#usage

```ts
const objectAssign = require('object-assign');
```

babel/system fudge this and make the default export the CJS module, but this is not accurate.

**Comment by mhegazy** [comment]
as noted, i do not think this is accurate change. please let me know why you believe this module should have a default export, otherwise, i am inclined to close it.

File: object-assign/object-assign.d.ts (hunk 1 above)
Comment (newash) [inline object-assign/object-assign.d.ts]: Well, according to this: [Typescrypt handbook/Modules](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require)
`export = objectAssign` is considered to be saying `objectAssign` is the default export.

Don't you agree?

**Comment by newash** [comment]
I added a comment underneath your inline note.
What do you think?

File: object-assign/object-assign.d.ts (hunk 1 above)
Comment (vvakame) [inline object-assign/object-assign.d.ts]: `import zip = require("./ZipCodeValidator");` is not same as `import zip from "./ZipCodeValidator";`.
`import zip = require("./ZipCodeValidator");` is CommonJS style.

**Comment by mhegazy** [comment]
As noted by @vvakame these two are not equivalent. and the library at run time does not seem to match the proposed declaration.

**Pull request closed without merging**